apps/easypid/src/hooks/useCredentialsWithCustomDisplay.ts

```diff
@@ -1,25 +1,51 @@
-import { type CredentialForDisplay, useCredentialsForDisplay } from 'packages/agent/src'
+import { type CredentialForDisplay, type CredentialForDisplayId, useCredentialsForDisplay } from '@package/agent'
+import { useMemo } from 'react'
 import { type PidSdJwtVcAttributes, usePidCredential } from './usePidCredential'
 
 type CustomCredentialForDisplay = CredentialForDisplay & {
-  attributesForDisplay: PidSdJwtVcAttributes
-  metadataForDisplay: Record<string, unknown>
+  attributesForDisplay?: PidSdJwtVcAttributes
+  metadataForDisplay?: Record<string, unknown>
 }
 
-export const useCredentialsWithCustomDisplay = () => {
-  const credentials = useCredentialsForDisplay()
-  const pidCredential = usePidCredential()
+export const useCredentialsWithCustomDisplayById = (id: CredentialForDisplayId) => {
+  const { credentials, isLoading } = useCredentialsWithCustomDisplay(true)
 
-  // replace PID credential with custom one
-  const index = credentials.credentials.findIndex(
-    (credential) => credential.metadata.type === pidCredential.credential?.type
-  )
-  if (index !== -1 && pidCredential.credential) {
-    credentials.credentials[index] = pidCredential.credential as CredentialForDisplay
+  return {
+    credential: useMemo(() => credentials.find((c) => c.id === id), [id, credentials]),
+    isLoading,
   }
+}
+
+export const useCredentialsWithCustomDisplay = (includeHiddenCredentials = false) => {
+  const { credentials, isLoading: isLoadingCredentialsForDisplay } = useCredentialsForDisplay()
+  const {
+    pidCredentialForDisplay: pidCredential,
+    credentialIds: pidCredentialIds,
+    credentials: pidCredentials,
+    isLoading: isLoadingPidCredential,
+  } = usePidCredential()
+
+  const filteredCredentials = useMemo(() => {
+    const pidIndex = credentials.findIndex((c) => c.id === pidCredential?.id)
+    const withoutPids = credentials
+      .filter((c) => !pidCredentialIds?.includes(c.id))
+      .map((c) => ({ ...c, isPid: false }))
+
+    // No pids, just return
+    if (pidIndex === -1 || !pidCredential) return withoutPids
+
+    // Insert only one of the pids
+    return [
+      ...withoutPids.slice(0, pidIndex),
+      ...(includeHiddenCredentials
+        ? pidCredentials.map((p) => ({ ...p, isPid: true }))
+        : [{ ...pidCredential, isPid: true }]),
```

Comment:
looks a bit hacky (and it is). But by default we don't include dupilcate PID, but if we are sharing the mdoc PID we do want to get it (so we trigger `includeHiddenCredentials`)

Reply:
Is the mdoc PID different than the sd-jwt PID? because that is why I initially didn't think this was needed and we can just reuse the display for the sd-jwt one while under the hood the mdoc one is shared.

Reply:
Maybe that is too much of an abstraction anyway so something like this had to happen

Reply:
yes the structure is different :/ So it would require quite some abstraction ( and mapping between keys).